src/Plotting/plot_recipes.jl

```diff
@@ -161,6 +162,56 @@ julia> plot(B, 1e-2)  # faster but less accurate than the previous call
         seriesalpha --> DEFAULT_ALPHA
         seriescolor --> DEFAULT_COLOR
 
+        # extract limit and extrema of plotted shapes if existings
+        lims = Dict()
+        extr = Dict()
+        p = plotattributes[:plot_object]
+        if length(p) > 0
+            subplot = p[1]
+            for symbol in [:x, :y]
+                lims[symbol] = subplot[Symbol(symbol,:axis)][:lims]
+                extr[symbol] = subplot[Symbol(symbol,:axis)][:extrema]
+            end
+        end
+
+        if !isbounded(X)
+            # if the limit is :auto, set the limits to the current extrema
+            for symbol in [:x, :y]
+                if lims[symbol] == :auto
```

Comment:
You need to do either of these:
* Add an `else` case to the first `if` and fill the dictionaries with default values there.
* Check below whether the dictionaries contain the keys before you access them.

(Or maybe you want a mixture: default values for `extr` and a check for `lims`?)

Otherwise there is an error (in a fresh session):
```julia
julia> using LazySets, Plots

julia> plot(HalfSpace([1., 1.], 1.))
ERROR: KeyError: key :x not found
```

Reply:
Thanks I added the default cases in the outsourced methods `_extract_limits` and `_extract_extrema` and used
```julia
xlims --> lims[:x]
ylims --> lims[:y]
```
to update the limits of the plot independent of `plotattributes[:plot_object]`

Reply:
Okay, does that not destroy the `:auto` for bounded sets now?

Reply:
No, because the default case in `_extract_limits` is `lims[:x] = (:auto); lims[:y] = (:auto)`

Reply:
If I understand correctly, that default is only used if there is no plot window yet.

Reply:
> If I understand correctly, that default is only used if there is no plot window yet.

I guess I did not understand correctly :smiley: